app/routes/u.github/update-profile.ts

```diff
@@ -0,0 +1,37 @@
+import { ActionFunction, redirect } from "@remix-run/node";
+import { updateProfileForUser } from "~/database.server";
+import { getCurrentUser, getSession } from "~/session.server";
+import { UserProfile } from "~/types";
+import { getProfileRouteForUser } from "~/utils/routes";
+
+export const action: ActionFunction = async ({ request }) => {
```

Comment:
This is where we handle the form submission from the "Welcome" modal. We parse the form data from the payload, send it to our database, and finally redirect to `/u/github/:slug`, where the user can see their fresh new profile.

Notice that this file doesn't export any React code, which means it's only an API route.

And that covers the authentication flow! Let's go over some of the important code for database operations.